cmd/share-secret/main.go

```diff
@@ -59,12 +64,30 @@ func PostSecret() http.HandlerFunc {
 	}
 }
 
-func GetSecret(w http.ResponseWriter, r *http.Request) {
+func GetSecret() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		key := mux.Vars(r)["key"]
+		if key == "" {
+			http.Error(w, "The path is not complete", http.StatusInternalServerError)
+			return
+		}
+
+		data := mData[key]
```

Comment:
Use this pattern instead
```
data, ok := mData[key]
```
If `ok` is false, we know there is no such element in the map. It is a safe check for nil pointers